code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm

```diff
@@ -9,119 +9,68 @@
-			C << "You are already reconstructing, or your body is currently recovering from the intense process of your previous reconstitution."
-			return
+/mob/living/carbon/human/proc/chimera_regenerate()
+	var/mob/living/carbon/human/C = src
+	var/nutrition_used = C.nutrition/2
 
-		if(C.stat == DEAD) //Uh oh, you died!
-			if(C.hasnutriment()) //Let's hope you have nutriment in you.... If not
-				var/time = (240+960/(1 + nutrition_used/75))
-				C.weakened = 10000 //Since it takes 1 tick to lose one weaken. Due to prior rounding errors, you'd sometimes unweaken before regenning. This fixes that.
-				C.reviving = 1
-				C.canmove = 0 //Make them unable to move. In case they somehow get up before the delay.
-				C << "You begin to reconstruct your form. You will not be able to move during this time. It should take aproximately [round(time)] seconds."
-				C.does_not_breathe = 1 //effectively makes them spaceworthy while regenning
-
-				for(var/obj/item/organ/E in C.bad_external_organs)
-					var/obj/item/organ/external/affected = E
-					for(var/datum/wound/W in affected.wounds) // Fix internal bleeds at the start of the rejuv process.
-						if(istype(W, /datum/wound/internal_bleeding))
-							affected.wounds -= W
-							affected.update_damages()
-
-				spawn(time SECONDS)
-					if(C) //Runtime prevention.
-						C << "<span class='notice'>Consciousness begins to stir as your new body awakens, ready to hatch..</span>"
-						C.verbs += /mob/living/carbon/human/proc/hatch
-						return
-					else
-						return //Something went wrong.
-			else //Dead until nutrition injected.
-				C << "Your body is too damaged to regenerate without additional nutrients to feed what few living cells remain."
-				return
+	if(C.reviving == 1) //If they're already unable to
+		C << "You are already reconstructing, or your body is currently recovering from the intense process of your previous reconstitution."
+		return
 
-		else if(C.stat != DEAD) //If they're alive at the time of reviving.
+	if(C.stat == DEAD) //Uh oh, you died!
+		if(C.hasnutriment()) //Let's hope you have nutriment in you.... If not
 			var/time = (240+960/(1 + nutrition_used/75))
 			C.weakened = 10000 //Since it takes 1 tick to lose one weaken. Due to prior rounding errors, you'd sometimes unweaken before regenning. This fixes that.
 			C.reviving = 1
 			C.canmove = 0 //Make them unable to move. In case they somehow get up before the delay.
 			C << "You begin to reconstruct your form. You will not be able to move during this time. It should take aproximately [round(time)] seconds."
 			C.does_not_breathe = 1 //effectively makes them spaceworthy while regenning
 
+			for(var/obj/item/organ/E in C.bad_external_organs)
+				var/obj/item/organ/external/affected = E
+				for(var/datum/wound/W in affected.wounds) // Fix internal bleeds at the start of the rejuv process.
+					if(istype(W, /datum/wound/internal_bleeding))
+						affected.wounds -= W
+						affected.update_damages()
```

Comment:
I dunno about this. It seems... shady. Those datums may or may not get GC'd, I don't know if they're referenced anywhere else. Maybe consider just healing them?

Reply:
Any suggestions how? This is how it's done in the equivalent changeling power.